it/proxy_test.py

```diff
@@ -0,0 +1,95 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import collections
+import os
+import shutil
+import tempfile
+
+import pytest
+
+import it
+from esrally.utils import process
+
+HttpProxy = collections.namedtuple("HttpProxy", ["authenticated_url", "anonymous_url"])
+
+
+@pytest.fixture(scope="module")
+def http_proxy():
+    config_dir = os.path.join(os.path.dirname(__file__), "resources", "squid")
+
+    lines = process.run_subprocess_with_output(f"docker run --rm --name squid -d "
+                                               f"-v {config_dir}/squidpasswords:/etc/squid/squidpasswords "
+                                               f"-v {config_dir}/squid.conf:/etc/squid/squid.conf "
+                                               f"-p 3128:3128 datadog/squid")
+    proxy_container_id = lines[0].strip()
+    proxy = HttpProxy(authenticated_url=f"http://testuser:testuser@127.0.0.1:3128",
+                      anonymous_url=f"http://127.0.0.1:3128")
+    yield proxy
+    process.run_subprocess(f"docker stop {proxy_container_id}")
+    process.run_subprocess(f"docker rm {proxy_container_id}")
```

Comment:
the `--rm` flag supplied to the `docker run` will take care of the docker remove when you stop it, and this ends up printing `"Error: No Such container:...."`

Reply:
I see. I think I just did a direct port of what we had in the integration test script and that's probably why I did not notice. I'll remove the `docker rm` invocation then.

Reply:
Addressed in https://github.com/elastic/rally/pull/1029/commits/daade765f8b865d58d80ec51b79c9c5633329887